### PR TITLE
[SYCL][E2E] Resolve reqd_sub_group_size issue for get_coordinate tests

### PR DESCRIPTION
**Pull request by YixingZhang007**
This PR resolves the reqd_sub_group_size issue and refactors the code in get_coordinate_ops_imp.hpp

### CODE REVIEW

File: sycl/test-e2e/Matrix/get_coordinate_ops_impl.hpp [hunk 1]
```diff
@@ -124,11 +122,7 @@ void test_get_coord_op() {
   TResult sum_cols[Cols] = {0};
   TResult sum_cols_ref[Cols] = {0};
 
-  for (int i = 0; i < Rows; i++) {
-    for (int j = 0; j < Cols; j++) {
-      M[i][j] = i + j;
-    }
-  }
+  matrix_fill(Rows, Cols, (T *)M, [](int i, int j) { return T(1) * (i + j); });
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/get_coordinate_ops_impl.hpp:125]:
```suggestion
  matrix_fill(Rows, Cols, (T *)M, [](int i, int j) { return T(i + j); });
```

**Comment by YixingZhang007** [comment]
This PR is for solving the Yury's comments mentioned in previous PR https://github.com/intel/llvm/pull/15880

File: sycl/test-e2e/Matrix/get_coordinate_ops_impl.hpp (hunk 1 above)
Comment (YixingZhang007) [reply sycl/test-e2e/Matrix/get_coordinate_ops_impl.hpp]: Thanks for the suggestion! I have fix this :)

**Review by YuriPlyakhin — APPROVED** [review]

**Comment by YixingZhang007** [comment]
Ping @intel/llvm-reviewers-runtime

**Comment by YixingZhang007** [comment]
@intel/llvm-gatekeepers This PR has been approved and all CI tests passed. Could you please help merge this PR? Thanks!

**Pull request merged**